File: 7/75/SortColors.java
/*
https://leetcode.com/problems/sort-colors/description/?envType=daily-question&envId=2025-05-17
*/

class Solution {
    public void sortColors(int[] nums) {
        int lengthOfNums=nums.length, countRed=0, countWhite=0, countBlue=0, beginIndex=0;
        for(int i=0; i<lengthOfNums; ++i){
            if(nums[i]==0) countRed++;
            else if(nums[i]==1) countWhite++;
            else countBlue++;
        }
        while(true){
            if(beginIndex>=lengthOfNums) break;
            while(countRed-- >0){
                nums[beginIndex++]=0;
                if(beginIndex>=lengthOfNums) break;
            }
            while(countWhite-- >0){
                nums[beginIndex++]=1;
                if(beginIndex>=lengthOfNums) break;
            }
            while(countBlue-- >0){
                nums[beginIndex++]=2;
                if(beginIndex>=lengthOfNums) break;
            }
        }
    }
}

File: 7/75/SortColors.cpp
/*
https://leetcode.com/problems/sort-colors/description/?envType=daily-question&envId=2025-05-17
*/

class Solution {
public:
    void sortColors(vector<int>& nums) {
        int sz=nums.size(), count0=0, count1=0, count2=0, idx=0;
        for(auto &vals: nums){
            if(vals==0) count0++;
            else if(vals==1) count1++;
            else count2++;
        }
        while(true){
            if(idx>=sz) break;
            while(count0--){
                nums[idx++]=0;
                if(idx>=sz) break;
            }
            while(count1--){
                nums[idx++]=1;
                if(idx>=sz) break;
            }
            while(count2--){
                nums[idx++]=2;
                if(idx>=sz) break;
            }
        }
    }
};